### ISSUE TITLE
Absolute class name must be supported

### ISSUE THREAD

**Issue by mvorisek** [markstory/sphinxcontrib-phpdomain#56]
There is namespace support like:
```
.. php:namespace:: Foo\Bar

.. php:class:: X
```

which renders `class Foo\Bar\X` correctly.

but
```
.. php:namespace:: Foo\Bar

.. php:class:: \Foo\Data\Y
```

support is broken. I would expect `Foo\Bar` ignored as the class name is fully qualified, ie. `class Foo\Data\Y` rendered.

The same for traits, references...

**Comment by markstory** [comment]
> I would expect Foo\Bar ignored as the class name is fully qualified, ie. class Foo\Data\Y rendered.

This isn't a scenario I had accounted for. Why are you mixing namespace directives with fully qualified class names in another namespace?

**Comment by mvorisek** [comment]
See https://github.com/atk4/ui/blob/3074d460793109f802e3542f6de0fdfec191d38a/docs/js.md?plain=1#L1 for example. The docs page mostly document `Js\Xxx` (`Atk4\Ui\Js\Xxx`) classes, but still need to refer to many `Yyy` (`Atk4\Ui\Yyy`) classes.

So I would expect `.. php:namespace::` to define: "the namespace for context below until redefined" and allow `.. php:class::` in form of `X` (unqualified name), `X\Sub` (`Sub` class name under global NS + `X` NS) and `\Foo\Y` (qualified name).

**Comment by mvorisek** [comment]
This issue seems tobe related with https://github.com/markstory/sphinxcontrib-phpdomain/pull/47 - what is the difference of `~` vs. the standartized leading `\`?

**Comment by markstory** [comment]
The `~` operator in role directives allows the classname to be omitted from link text.

**Comment by mvorisek** [comment]
@markstory I saw your https://github.com/markstory/sphinxcontrib-phpdomain/commit/3e98837cd56c47495f27aa7db549e958b840310e.

Might I know what exactly was wrong in #60?

Also might I know what are your preffered communication channels as #65 intended to fix the CI but 2 years there was no reactions.

And I do not belive this issue is fixed correctly as https://github.com/markstory/sphinxcontrib-phpdomain/commit/4f343e88f0128d8e8f69ad3218c655b8feaca4ec changes the current NS, but this is wrong, as any following class /wo `\` will have wrong NS.

**Comment by mvorisek** [comment]
Mark, it hurst that I have to see all this because of "cron" and I feel at least https://github.com/markstory/sphinxcontrib-phpdomain/commit/3e98837cd56c47495f27aa7db549e958b840310e is restoring many quirks I was nailing many days as they were producing very unexpected results...

My question is what was exactly wrong. At least I do not understand the reason of #70 issue from the issue itself.

As https://github.com/markstory/sphinxcontrib-phpdomain/commit/3e98837cd56c47495f27aa7db549e958b840310e reverted many fixes and CLI warnings (emit by `log_info()`), I would say some testing is ⚠ broken or even removed.

Because of this, I would ask you to reconsider reverting the master branch to `6e244a1` and then to fix original CI /wo any code changes. Then the fixes can be landed safely.

**Comment by markstory** [comment]
> I would ask you to reconsider reverting the master branch to 6e244a1

I'm not going to do that, as it will re-introduce the regressions with cross linking. In my own projects we were facing hundreds of warnings because of breaking changes in cross linking. This extension has behaved fairly consistently for many years, and changing that behavior needlessly introduces maintenance toil.

If there are remaining problems with cross linking or object definitions, I'm happy to discuss those problems and tactical solutions to those problems.

**Comment by mvorisek** [comment]
The warnings have a purpose! - notify the user that somethings is wrong. As newly the warnings are no longer present, even wrong syntax/targets are accepted. What is the advantage?

In atk4/data I went even further and baselined the warnings - https://github.com/atk4/data/blob/develop/docs/baseline.txt - so there are known problems in the docs and the CI asserts they shink only.

> In my own projects we were facing hundreds of warnings because of breaking changes in cross linking. This extension has behaved fairly consistently for many years, and changing that behavior needlessly introduces maintenance toil.

Maybe we can find some solution to the "breaking changes" you have?

**Comment by markstory** [comment]
> The warnings have a purpose! - notify the user that somethings is wrong.

What is wrong though? The links resolved correctly in the past, and now again.  It is intentional that within a class/namespace, links inherit the current class/namespace scope. Those links no longer resolving and emitting warnings was the bulk of the breaking changes that I faced.

**Comment by mvorisek** [comment]
Definitely! The changes I landed definitely honor the currect class/namespace, thus the question, what is no longer compatible or wrong.

Maybe we can release 1.x with the changes I made and keep the revert for 0.x.
